io-engine/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -1017,8 +1052,42 @@ impl<'n> Nexus<'n> {
     /// with the nexus paused once they are awakened via resume().
     /// Note: in order to handle concurrent pauses properly, this function must
     /// be called only from the master core.
-    pub async fn pause(self: Pin<&mut Self>) -> Result<(), Error> {
-        self.io_subsystem_mut().suspend().await
+    pub async fn pause(mut self: Pin<&mut Self>) -> Result<(), Error> {
+        EventWithMeta::event(
+            self.deref(),
+            EventAction::SubsystemPause,
+            subsystem_pause_event_meta("started", None, None),
+        )
+        .generate();
+        let start_time = std::time::Instant::now();
+        let result = self.as_mut().io_subsystem_mut().suspend().await;
+        match result {
+            Ok(_) => {
+                EventWithMeta::event(
+                    self.deref(),
+                    EventAction::SubsystemPause,
+                    subsystem_pause_event_meta(
+                        "completed",
```

Comment:
"completed" or "failed" seems to be unnecessary the way it;s used here, since we can infer that from the error.
I think including the actual subsystem status would be better here.

Reply:
Yes. I added NexusPauseState now.